assets/js/util/i18n.js

```diff
@@ -19,22 +19,189 @@
 /**
  * External dependencies
  */
-import { get } from 'lodash';
+import { get, isFinite } from 'lodash';
 
 /**
  * WordPress dependencies
  */
-import { __ } from '@wordpress/i18n';
+import { __, sprintf } from '@wordpress/i18n';
+
+/**
+ * Converts seconds to a display ready string indicating
+ * the number of hours, minutes and seconds that have elapsed.
+ *
+ * For example, passing 65 returns '1m 5s'.
+ *
+ * @since 1.0.0
+ *
+ * @param {number} seconds The number of seconds.
+ * @return {string} Human readable string indicating time elapsed.
+ *
+ */
+export const prepareSecondsForDisplay = ( seconds ) => {
+	seconds = parseInt( seconds, 10 );
+
+	if ( isNaN( seconds ) || 0 === seconds ) {
+		return '0.0s';
+	}
+	const results = {};
+	results.hours = Math.floor( seconds / 60 / 60 );
+	results.minutes = Math.floor( ( seconds / 60 ) % 60 );
+	results.seconds = Math.floor( seconds % 60 );
+
+	const returnString =
+		( results.hours ? results.hours + 'h ' : '' ) +
+		( results.minutes ? results.minutes + 'm ' : '' ) +
+		( results.seconds ? results.seconds + 's ' : '' );
```

Comment:
@felixarntz Perhaps beyond the scope here, but the formatting of time should use `numberFormat` (or `numFmt`) with `style: 'unit', unitDisplay: 'narrow'` and `unit: timeUnit` where `timeUnit` is one of `second`, `minute`, or `hour`.

This would result in the same output, but properly localized to use the correct abbreviations for the unit of time.

Unfortunately there doesn't seem to be a single function in `Intl` for formatting an amount of time in this way using multiple units so the way they are concatenated should probably be translatable as well.

Reply:
Maybe worth a follow-up issue?